### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by e-gloo**
Login and Register features
#3mg41n #3mg4b0 #3wkezb #3wkez3 #3wkg6n

### CODE REVIEW

**Comment by e-gloo** [comment]
#3mg41n #3mg4b0 #3wkezb #3wkez3 #3wkg6n

File: app/Helpers/Authentication.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const authToken = Env.get('TWILIO_TOKEN')
+const helePhone = Env.get('TWILIO_PHONE')
+const twilio = require('twilio')
+
+const generatePassword = (length = 10) => {
+  return (Math.random().toString(36).substr(2, 1 + length / 2) +
+    Math.random().toString(36).substr(2, 1 + length / 2)).slice(-length)
+}
+
+/* istanbul ignore next */
```
Comment (LeNiglo) [inline app/Helpers/Authentication.js:16]: Why do you need this ?
Only for the naming convention of the function ?
SMS is only for French, it should be sendText or sendMessage in my opinion.

File: database/migrations/1582752578420_user_schema.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      table.string('city', 100)
+      table.string('phone_pro', 20)
+      table.bool('active').defaultTo(true).index()
+      table.datetime('last_login').defaultTo(this.fn.now())
+      table.timestamps()
+    })
+  }
+
+  down() {
+    this.drop('users')
```
Comment (LeNiglo) [inline database/migrations/1582752578420_user_schema.js:30]: Is it not required to drop any foreign keys ?

**Review by LeNiglo — COMMENTED** [review]
I haven't looked at the unit tests, but I trust you.

File: database/migrations/1582752578420_user_schema.js (hunk 2 above)
Comment (e-gloo) [reply database/migrations/1582752578420_user_schema.js:30]: No. Once a user is registered, even if the establishment is deleted, the user should remain.

File: app/Helpers/Authentication.js (hunk 1 above)
Comment (e-gloo) [reply app/Helpers/Authentication.js:16]: Is it just the function name that is bothering you or the function itself?

Comment (LeNiglo) [reply app/Helpers/Authentication.js:16]: It is the fact that you need to disable the linter only for naming a function.
```
/* istanbul ignore next */
```
We should adapt ourselves to the linter and not force the linter to adapt to us, right ?
I will quickly rename sendSMS and the corresponding references to sendTextMessage if you are agree.

Comment (e-gloo) [reply app/Helpers/Authentication.js:16]: Istanbul is the code coverage, not the linter.
The exception is temporary, I need to mock the sms in the unit tests, so for now it is not tested.

**Review by LeNiglo — APPROVED** [review]

**Pull request merged**